csharp/src/Drivers/Databricks/DatabricksCompositeReader.cs

```diff
@@ -108,5 +110,42 @@ private BaseDatabricksReader DetermineReader(TFetchResultsResp initialResults)
 
             return await _activeReader.ReadNextRecordBatchAsync(cancellationToken);
         }
+
+        public override async ValueTask<RecordBatch?> ReadNextRecordBatchAsync(CancellationToken cancellationToken = default)
+        {
+            var result = await ReadNextRecordBatchInternalAsync(cancellationToken);
+            // Stop the poller when we've reached the end of results
+            if (result == null)
+            {
+                StopOperationStatusPoller();
+            }
+            return result;
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _activeReader?.Dispose();
+                DisposeOperationStatusPoller();
+            }
+            _activeReader = null;
+            base.Dispose(disposing);
+        }
+
+        private void DisposeOperationStatusPoller()
+        {
+            if (operationStatusPoller != null)
+            {
+                StopOperationStatusPoller();
+                operationStatusPoller.Dispose();
+                operationStatusPoller = null;
+            }
+        }
+
+        private void StopOperationStatusPoller()
+        {
+            operationStatusPoller?.Stop();
```

Comment:
Consider setting to `null` here instead of the `DisposeOperationStatusPoller` method to avoid duplicate calls.